scripts/sentiment_analysis/sentiment_analysis_cnn.py

```diff
@@ -71,138 +95,166 @@
-        total_sample_num += label.shape[0]
-        total_correct_num += (pred.astype('int') == label).sum().asscalar()
-        if (i + 1) % args.log_interval == 0:
-            print('[Batch {}/{}] elapsed {:.2f} s'.format(
-                i + 1, len(dataloader), time.time() - start_log_interval_time))
-            start_log_interval_time = time.time()
-    avg_L = total_L / float(total_sample_num)
-    acc = total_correct_num / float(total_sample_num)
-    return avg_L, acc
-
-def train(net, train_data, test_data, dev_data=None):
-    """Train textCNN model for sentiment analysis."""
-    start_pipeline_time = time.time()
-    net, trainer = text_cnn.init(net, vocab, args.model_mode, context)
-    if dev_data is None:
-        random.shuffle(train_data)
-        sp = len(train_data) // 10
-        train_dataloader = DataLoader(dataset=train_data[sp:],
-                                      batch_size=args.batch_size,
-                                      shuffle=True)
-        val_dataloader = DataLoader(dataset=train_data[:sp],
-                                    batch_size=args.batch_size,
-                                    shuffle=False)
+
+def check_metrics(metrics):
+    if isinstance(metrics, mx.metric.CompositeEvalMetric):
+        metrics = [
+            m for metric in metrics.metrics for m in check_metrics(metric)
+        ]
+    elif isinstance(metrics, mx.metric.EvalMetric):
+        metrics = [metrics]
     else:
-        train_dataloader = DataLoader(dataset=train_data,
+        metrics = metrics or []
+        if not all(
+                [isinstance(metric, mx.metric.EvalMetric) for metric in metrics]):
+            raise ValueError(
+                "metrics must be a Metric or a list of Metric, "
+                "refer to mxnet.metric.EvalMetric: {}".format(metrics))
+    return metrics
+
+
+class GetValMetricHandler(EpochBegin, BatchEnd, EpochEnd):
+    def __init__(self, metrics):
+        self.metrics = check_metrics(metrics)
+        self.metric_history = {}
+
+    def epoch_begin(self, estimator, *args, **kwargs):
+        for metric in self.metrics:
+            metric.reset()
+
+    def batch_end(self, estimator, *args, **kwargs):
+        pred = kwargs['pred']
+        label = kwargs['label']
+        loss = kwargs['loss']
+        for metric in self.metrics:
+            if isinstance(metric, mx.metric.Loss):
+                # metric wrapper for loss values
+                metric.update(0, loss)
+            else:
+                metric.update(label, pred)
+
+    def epoch_end(self, estimator, *args, **kwargs):
+        for metric in self.metrics:
+            metric_name, metric_val = metric.get()
+            if 'validation' in metric_name:
+                self.metric_history.setdefault(metric_name,
+                                               []).append(metric_val)
+
+
+class SentimentAnalysisCNNBatchProcessor(BatchProcessor):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
```

Comment:
I believe the default batch processor does not have any arguments and it is constructor is just a pass, why do you call the super class constructor here?